src/edu/psu/sweng500/team8/gui/CustomKeyListener.java

```diff
@@ -24,7 +24,10 @@ public CustomKeyListener(Cell cell, GameSession gameSession) {
 
 	public void keyTyped(java.awt.event.KeyEvent keyEvent) {
 		
-		if (KeyEvent.VK_BACK_SPACE == keyEvent.getKeyChar()) {
+		
+		
+		if (KeyEvent.VK_BACK_SPACE == keyEvent.getKeyChar() || KeyEvent.VK_DELETE == keyEvent.getKeyChar()) {
+			
 			gameSession.enterNumber(this.cell, 0);
 		} else {
 			gameSession.enterNumber(this.cell, Integer.parseInt(String.valueOf(keyEvent.getKeyChar())));

```

Comment:
Yes, KeyListener will always be called first. We can call validation logic
from here to avoid exception.

On Tue, Jul 7, 2015 at 9:08 AM, Jeremy Nelson <notifications@github.com>
wrote:

> In src/edu/psu/sweng500/team8/gui/CustomKeyListener.java
> <https://github.com/garnryang/Team8/pull/10#discussion_r34036714>:
>
> > @@ -24,7 +24,10 @@ public CustomKeyListener(Cell cell, GameSession gameSession) {
> >
> >  	public void keyTyped(java.awt.event.KeyEvent keyEvent) {
> >  		
> > -		if (KeyEvent.VK_BACK_SPACE == keyEvent.getKeyChar()) {
> > +		
> > +		
> > +		if (KeyEvent.VK_BACK_SPACE == keyEvent.getKeyChar() || KeyEvent.VK_DELETE == keyEvent.getKeyChar()) {
> > +			
> >  			gameSession.enterNumber(this.cell, 0);
> >  		} else {
> >  			gameSession.enterNumber(this.cell, Integer.parseInt(String.valueOf(keyEvent.getKeyChar())));
>
> David's logic checks the key to ensure it is a valid integer. If this
> event handler is hit before David's, or if both event handlers are called,
> this will throw an exception if an invalid character is entered.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/garnryang/Team8/pull/10/files#r34036714>.
>